File: 2021W28/B0neh3ad/pgr_42586.cpp
/*
    N으로 표현
    https://programmers.co.kr/learn/courses/30/lessons/42895
*/
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int front_idx = 0;
    while(front_idx < progresses.size()){
        int cnt = 0;
        for(int i=front_idx; i<progresses.size(); i++){ progresses[i] += speeds[i]; }
        for(int i=front_idx; (i<progresses.size()) && (progresses[i] >= 100); i++, cnt++){}
        if(cnt > 0){
            answer.push_back(cnt);
            front_idx += cnt;
        }
    }
    return answer;
}

File: 2021W28/fienestar/pgr_42586.js
/*
    기능개발
    https://programmers.co.kr/learn/courses/30/lessons/42586
*/
function solution(progresses, speeds) {
    let answer = [];
    let day = -1

    for(let i in progresses){
        let need = Math.ceil((100-progresses[i])/speeds[i])
        if(need <= day)
            answer[answer.length - 1]++
        else{
            answer.push(1)
            day = need
        }
    }
    
    return answer;
}